src/neurostatslib/glm.py

```diff
@@ -1,431 +1,598 @@
         ValueError
-            If attempting to simulate a different number of neurons than were
-            present during fitting (i.e., if ``init_spikes.shape[0] !=
-            self.baseline_log_fr_.shape[0]``).
-        UserWarning
-            If there are any zeros in ``self.predict(spike_data)``, since this
-            will likely lead to infinite log-likelihood values being returned.
+            - If the instance has not been previously fitted.
+            - If there's an inconsistency between the number of neurons in model parameters.
+            - If the number of neurons in input arguments doesn't match with model parameters.
 
+
+        See Also
+        --------
+        [predict](./#neurostatslib.glm.GLM.predict) :
+        Method to predict rates based on the model's parameters.
         """
-        # ignore the last time point from predict, because that corresponds to
-        # the next time step, which we have no observed data for
-        predicted_firing_rates = self.predict(spike_data)[:, :-1]
-        if (predicted_firing_rates == 0).any():
-            warnings.warn(
-                "predicted_firing_rates array contained zeros, this can "
-                "lead to infinite log-likelihood values."
-            )
-        window_size = self.spike_basis_matrix.shape[1]
-        return self._score(predicted_firing_rates, spike_data[:, window_size:])
+        # check if the model is fit
+        self._check_is_fit()
+        Ws, bs = self.basis_coeff_, self.baseline_link_fr_
+        (feedforward_input,) = self.preprocess_simulate(
+            feedforward_input, params_f=(Ws, bs)
+        )
+        predicted_rate = self._predict((Ws, bs), feedforward_input)
+        return (
+            self.noise_model.emission_probability(
+                key=random_key, predicted_rate=predicted_rate
+            ),
+            predicted_rate,
+        )
 
-    def simulate(
+
+class GLMRecurrent(GLM):
+    """
+    A Generalized Linear Model (GLM) with recurrent dynamics.
+
+    This class extends the basic GLM to capture recurrent dynamics between neurons,
+    making it more suitable for simulating the activity of interconnected neural populations.
+    The recurrent GLM combines both feedforward inputs (like sensory stimuli) and past
+    neural activity to simulate or predict future neural activity.
+
+    Parameters
+    ----------
+    noise_model : nsm.NoiseModel, default=nsm.PoissonNoiseModel()
+        The noise model to use for the GLM. This defines how neural activity is generated
+        based on the underlying firing rate. Common choices include Poisson and Gaussian models.
+
+    solver : slv.Solver, default=slv.RidgeSolver()
+        The optimization solver to use for fitting the GLM parameters.
+
+    data_type : {jnp.float32, jnp.float64}, optional
+        The numerical data type for internal calculations. If not provided, it will be inferred
+        from the data during fitting.
```

Comment:
```suggestion
    noise_model : 
        The noise model to use for the GLM. This defines how neural activity is generated
        based on the underlying firing rate. Common choices include Poisson and Gaussian models.
    solver : 
        The optimization solver to use for fitting the GLM parameters.
```
don't document types or defaults in docstring, they're automatically parsed from function signature